### DISCUSSION TITLE
ArgoCD with Helm Umbrella Chart

### DISCUSSION THREAD

**Discussion by michalip** [argoproj/argo-cd#5296] [Q&A]
Hello, 
I'm interested in using ArgoCD but I can't seem to find info on Helm umbrella charts + argo + gitops.

The scenario requires 60+ microservices to be deployed/updated to K8s
To satisfy this need we have an umbrella Helm chart, listing all the microservices as SUBCHARTS and using _tpl inheritance in Helm we are able to install all subcharts under the umbrella chart in one go.

The microservices have their own git repos (bitbucket) , 60 repos.
The umbrella helm chart will also have its own repo and since this is a single chart, it will be only in 1 repo, not 60.
Any dev can go and do a PR -> umbrella chart update 1 microservice to a new image

The question is will Argo detect an app/service/subchart  being out of sync if we are using an umbrella chart ?
Will it display one app or 60 apps ? 
Can this work in Argo or do I need 60 individual helm charts for each microservice repo?

Any one have a similar setup with an umbrella app or a doc for this?

Much appreciated

**Comment by kshamajain99** [comment]
> Any one have a similar setup with an umbrella app or a doc for this?

You can refer argocd example apps https://github.com/argoproj/argocd-example-apps/blob/master/helm-dependency/README.md

Other resources I found on web:
https://www.arthurkoziel.com/setting-up-argocd-with-helm/
https://www.reddit.com/r/devops/comments/l2rb1y/helm_umbrella_charts_and_argocdflux/?utm_source=share&utm_medium=web2x&context=3
